From b693079332152e09b22724c26efd5acac8160472 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Wed, 12 Dec 2018 06:21:29 +0530
Subject: [PATCH 100/255] octeontx2-af: Don't treat lack of CGX interfaces as
 error

commit ba8c5e3e42263e8d0145049bd7885b791f0ad2c5 from
git@git.assembla.com:cavium/WindRiver.linux.git

Don't treat lack of CGX LMACs on the system as a error.
Instead ignore it so that atleast LBK VFs are created
and can be used.

Change-Id: I4bd005d5371f6449d0189ab12809f17aa7cf8742
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_cgx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cgx.c
index 7d7133c5f799..a5dc8445a117 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cgx.c
@@ -272,7 +272,7 @@ int rvu_cgx_init(struct rvu *rvu)
 	rvu->cgx_cnt_max = cgx_get_cgxcnt_max();
 	if (!rvu->cgx_cnt_max) {
 		dev_info(rvu->dev, "No CGX devices found!\n");
-		return -ENODEV;
+		return 0;
 	}
 
 	rvu->cgx_idmap = devm_kzalloc(rvu->dev, rvu->cgx_cnt_max *
-- 
2.17.1

